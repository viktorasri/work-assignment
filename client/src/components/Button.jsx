import React from 'react';
import PropTypes from 'prop-types';

const Button = ({ children, variant, ...props }) => {
  return (
    <button className={`button ${variant ? `button--${variant}` : ''}`} {...props}>
      {children}
    </button>
  );
};

Button.propTypes = {
  children: PropTypes.oneOfType([PropTypes.element, PropTypes.string]).isRequired,
  variant: PropTypes.oneOf(['primary', 'icon']),
};

export default Button;
