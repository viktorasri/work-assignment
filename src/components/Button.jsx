import React from 'react';
import PropTypes from 'prop-types';

const Button = ({ children, variant, ...props }) => {
  return (
    <button className={`button ${variant ? `button--${variant}` : ''}`} {...props}>
      {children}
    </button>
  );
};

Button.propTypes = {
  children: PropTypes.oneOfType([PropTypes.element, PropTypes.string]).isRequired,
  variant: PropTypes.oneOf(['primary', 'secondary', 'icon']),
};

export default Button;
