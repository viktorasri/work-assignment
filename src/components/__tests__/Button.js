import React from 'react';
import { shallow } from 'enzyme';
import Button from '../Button';

describe('<Button />', () => {
  const wrapper = shallow(<Button variant='primary'>button</Button>);

  it('expect to render component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('expect to render without variant', () => {
    expect(shallow(<Button>button</Button>)).toMatchSnapshot();
  });

  it('expect to pass a variant of "primary"', () => {
    expect(wrapper.find('.button--primary')).toHaveLength(1);
  });
});
