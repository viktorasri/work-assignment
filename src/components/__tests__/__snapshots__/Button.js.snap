// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> expect to render component 1`] = `
<button
  className="button button--primary"
>
  button
</button>
`;

exports[`<Button /> expect to render without variant 1`] = `
<button
  className="button "
>
  button
</button>
`;
